dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueChannel.java

```diff
@@ -31,137 +30,28 @@
-        this.nodeSelectionChannel = NodeSelectionStrategyChannel.create(cf);
-        this.queuedChannel = QueuedChannel.create(cf, nodeSelectionChannel);
-        this.delegate = wrapQueuedChannel(cf, queuedChannel);
-        updateUrisInner(cf.clientConf().uris(), true);
-    }
-
-    private static ImmutableConfig withUris(Config cf, List<String> elements) {
-        return ImmutableConfig.builder()
-                .from(cf)
-                .rawConfig(ClientConfiguration.builder()
-                        .from(cf.clientConf())
-                        .uris(elements)
-                        .build())
-                .build();
-    }
-
-    private static LimitedChannel createPerUriChannel(Config cf, String uri) {
-        Channel channel = cf.channelFactory().create(uri);
-        // Instrument inner-most channel with instrumentation channels so that we measure only the over-the-wire-time
-        channel = new InstrumentedChannel(
-                channel, cf.channelName(), cf.clientConf().taggedMetricRegistry());
-        channel = HostMetricsChannel.create(channel, cf, uri);
-        channel = new ActiveRequestInstrumentationChannel(
-                channel, cf.channelName(), "running", cf.clientConf().taggedMetricRegistry());
-        // TracedChannel must wrap TracedRequestChannel to ensure requests have tracing headers.
-        channel = new TraceEnrichingChannel(channel);
-
-        ChannelToLimitedChannelAdapter limited = new ChannelToLimitedChannelAdapter(channel);
-        return ConcurrencyLimitedChannel.create(
-                cf, limited, cf.clientConf().uris().indexOf(uri));
-    }
-
-    private static EndpointChannelFactory wrapQueuedChannel(Config cf, QueuedChannel queuedChannel) {
-        return endpoint -> {
-            EndpointChannel channel = new EndpointChannelAdapter(endpoint, queuedChannel);
-            channel = new TracedChannel(channel, "Dialogue-request-attempt");
-            channel = RetryingChannel.create(cf, channel, endpoint);
-            channel = UserAgentEndpointChannel.create(
-                    channel, endpoint, cf.clientConf().userAgent().get());
-            channel = DeprecationWarningChannel.create(cf, channel, endpoint);
-            channel = new ContentDecodingChannel(channel);
-            channel = DialogueTracedRequestChannel.create(channel, endpoint);
-            channel = ActiveRequestInstrumentationChannel.create(cf, channel, endpoint, "processing");
-            return NeverThrowChannel.create(channel); // this must come last as a defensive backstop
-        };
+    private DialogueChannel(Config cf, EndpointChannelFactory delegate) {
+        this.cf = cf;
+        this.delegate = delegate;
     }
 
     @Override
     public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
         return delegate.endpoint(endpoint).execute(request);
     }
 
-    public void updateUris(Collection<String> uris) {
```

Comment:
@carterkozak @ferozco I could leave this method around by just delete the method body to avoid any danger of ABI break? what do you think?

Reply:
I think it would be worse if it was used and didn't do anything than if it disappears.